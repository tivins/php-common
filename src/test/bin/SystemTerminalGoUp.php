<?php

use Tivins\Core\Chrono;
use Tivins\Core\System\Terminal;

require 'vendor/autoload.php';

$array = [
    'part1' => ['action1', 'action2', 'action3'],
    'part2' => ['action4', 'action5'],
];

$chrono = (new Chrono())->start();
foreach ($array as $partName => $actions) {
    $actionChrono = (new Chrono())->start();
    echo " ðŸ”§ $partName\n";
    foreach ($actions as $action) {
        echo " â””â”€ processing $action ...\n";
        usleep(rand(400000, 600000)); // fake processing
        Terminal::goUpClean(1);
    }
    Terminal::goUpClean(1);
    echo " âœ… $partName (" . number_format($actionChrono->get(), 2) . " s.)\n";
}
echo "All tasks finished in " . number_format($chrono->get(), 2) . " seconds.\n";