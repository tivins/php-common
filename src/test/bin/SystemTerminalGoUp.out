 ðŸ”§ part1
 â””â”€ processing action1 ...
[1A[K â””â”€ processing action2 ...
[1A[K â””â”€ processing action3 ...
[1A[K[1A[K âœ… part1 (1.54 s.)
 ðŸ”§ part2
 â””â”€ processing action4 ...
[1A[K â””â”€ processing action5 ...
[1A[K[1A[K âœ… part2 (0.96 s.)
All tasks finished in 2.50 seconds.
